### PR TITLE
[INTERNAL] add binary operator skeleton for scoped enums

### PR DESCRIPTION
**Pull request by h-2**

### CODE REVIEW

File: test/unit/core/enum_binary_ops_test.cpp [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    my_enum e2 = e ^ my_enum::VAL2;
+    EXPECT_EQ(e2, my_enum::COMB);
+}
+
+TEST(add_enum_binary_ops, NOT)
+{
+    my_enum e = my_enum::VAL1;
+    my_enum e2 = e;
+    ~e2;
+    ~e2;
```
Comment (marehr) [inline test/unit/core/enum_binary_ops_test.cpp]: this tests nothing; could you assign the `~` results?

File: include/seqan3/core/enum_binary_ops.hpp [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+ * constexpr bool seqan3::add_enum_binary_ops<my_enum> = true;
+ *
+ * my_enum e = my_enum::VAL1;
+ * my_enum e2 = e | my_enum::VAL2;
+ *
+ * // e2 == my_enum::COMB;
+ * ```
+ */
+template <typename t>
+constexpr bool add_enum_binary_ops = false;
```
Comment (marehr) [inline include/seqan3/core/enum_binary_ops.hpp]: Can you change it to `add_enum_bitwise_operators`? We wanted to write stuff out.

Also `~`is a unary operator and not a binary operator. I would suggest `bitwise` operators.

File: test/unit/core/enum_binary_ops_test.cpp [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    EXPECT_EQ(e, e2);
+}
```
Comment (marehr) [inline test/unit/core/enum_binary_ops_test.cpp]: The assignment versions are not tested: `|=`, `^=`, `&=`

File: test/unit/core/enum_binary_ops_test.cpp (hunk 1 above)
Comment (h-2) [reply test/unit/core/enum_binary_ops_test.cpp]: ROFL

**Comment by h-2** [comment]
@marehr The documentation can't be generated automatically, because enums aren't treated as real types, therefore they can't satisfy concepts ("implement interfaces") and you cannot `\relates` to them. If write an enum and overload the metafunction, just mention this in your documentation.

I think I have addressed all your other points.

File: include/seqan3/core/add_enum_bitwise_operators.hpp [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+/*!\name Binary operators for scoped enums
+ * \brief Perform binary operations like on ints or weak enums. These overloads are available if
+ * seqan3::add_enum_bitwise_operators is defined for your type.
+ * \ingroup core
+ * \see seqan3::add_enum_bitwise_operators
+ * \{
+ */
+template <typename t>
+constexpr t operator& (t lhs, t rhs)
```
Comment (rrahn) [inline include/seqan3/core/add_enum_bitwise_operators.hpp]: it would be good to define them noexcept, or is this done implicitly here?

Comment (h-2) [reply include/seqan3/core/add_enum_bitwise_operators.hpp]: yes, I have just added that

**Pull request merged**